src/test_l3_dsHost.c

```diff
@@ -0,0 +1,325 @@
+ */
+
+
+/**
+ * @file test_l2_dsAudio.c
+ *
+ */
+#include <ut.h>
+#include <ut_log.h>
+#include <ut_kvp_profile.h>
+#include "dsHost.h"
+#include "test_parse_configuration.h"
+
+#define DSHOST_SOC_LENGTH    20
+#define DS_HOST_KVP_SIZE     128
+
+static int gTestGroup = 2;
+static int gTestID = 1;
+
+int prevTemp = 0;
+
+
+/* dsError_t */
+const static ut_control_keyStringMapping_t dsError_mapTable [] = {
+  { "dsERR_NONE",                    (int32_t)dsERR_NONE                    },
+  { "dsERR_GENERAL",                 (int32_t)dsERR_GENERAL                 },
+  { "dsERR_INVALID_PARAM",           (int32_t)dsERR_INVALID_PARAM           },
+  { "dsERR_INVALID_STATE",           (int32_t)dsERR_INVALID_STATE           },
+  { "dsERR_ALREADY_INITIALIZED",     (int32_t)dsERR_ALREADY_INITIALIZED     },
+  { "dsERR_NOT_INITIALIZED",         (int32_t)dsERR_NOT_INITIALIZED         },
+  { "dsERR_OPERATION_NOT_SUPPORTED", (int32_t)dsERR_OPERATION_NOT_SUPPORTED },
+  { "dsERR_RESOURCE_NOT_AVAILABLE",  (int32_t)dsERR_RESOURCE_NOT_AVAILABLE  },
+  { "dsERR_OPERATION_FAILED",        (int32_t)dsERR_OPERATION_FAILED        },
+  { "dsErr_MAX",                     (int32_t)dsErr_MAX                     },
+  {  NULL, -1 }
+};
+
+/**
+* @brief Initialization of the HAL dsHost Module
+*
+* This test provides a scope to open the HAL dsHost module.
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 001@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_dsHost_hal_Init(void)
+{
+   gTestID = 1;
+   dsError_t status = dsERR_NONE ;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call dsHostInit()
+   status = dsHostInit();
+   UT_LOG_INFO("Result dsHostInit: dsError_t:[%s]",
+                 UT_Control_GetMapString(dsError_mapTable, status));
+
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief getTemperature of the HAL dsHost Module
+*
+* This test provides a scope to test the get Temperature function.
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 002@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_dsHost_hal_get_Temperature(void)
+{
+   gTestID = 2;
+   dsError_t status = dsERR_NONE ;
+
+   float cpuTemperature = 0;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call dsGetCPUTemperature()
+   status = dsGetCPUTemperature(&cpuTemperature);
+   UT_LOG_INFO("Result dsGetCPUTemperature: dsError_t:[%s]",
+                 UT_Control_GetMapString(dsError_mapTable, status));
+    UT_LOG_INFO("CPU Temperature: %f\n", cpuTemperature);
+
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+
+   if(prevTemp == 0)
```

Comment:
in L3 verification need to implement in Python

Reply:
What do you mean?

Reply:
test_l3_dsHost_hal_get_Temperature(C code) function just prints the current temperature, user/python calls  test_l3_dsHost_hal_get_Temperature multiple times(using the menu) and decides whether the temperature raise is within the accepted limit or not

Reply:
Updated as requested